### ISSUE TITLE
Folder Import Should Also Be Multi-Item Selected Import

### ISSUE THREAD

**Issue by AmasterAmaster** [Yellow-Dog-Man/Resonite-Issues#2692]
### Is your feature request related to a problem? Please describe.

I know this already works, but it makes a bunch of raw files. For new users trying to get avatars imported, they get confused when mentors say to "bring in your unity files", "Drag and drop your avatar and textures", or similar language, which makes them think to just drag the entire folder into Resonite. This in turn makes the process so bad because now there is a bunch of raw files everywhere in the world and the new user gets frustrated as they cant see their avatar model or their textures.

![omg why 01](https://github.com/user-attachments/assets/b223a440-62a9-4bf6-bd2e-8145012c8ee6)

Another idea someone suggested in the discord discussion is when a user drags a folder into Resonite, that it could just bring in all the important stuff like .FBX or .PNGs and import those all at once, just to make it easier for the new users.

### Describe the solution you'd like

To streamline the process for new users (especially those that don't know the platform and are just getting into it), dragging a folder should have the same ability as multi selecting items from the windows file explorer and dragging those selected items into resonite (basically treating it the same).

### Describe alternatives you've considered

- Digging through the persons computer and invading their privacy (definitely a safety concern).
- Going through each of the spawned raw files one by one, and dealing with the slow process.
- Having the new user go through their computer one file at a time (a slow and painful process).

### Additional Context

I know it is just a few clicks more to actually import the files from those raw files, it is just jarring for new users when they are bombarded with a bunch of things and not the actual things that they care about.

### Requesters

AmasterAmaster

**Comment by ko-tengu** [comment]
Yeah, I'm in support of the batch importer defaulting to only importing in "usable" files instead of importing in raw files items that most users won't have a use for. It could be a checkbox at the bottom of the window, or options for raw file imports in a submenu similar to the single item importer.

**Comment by ProbablePrime** [comment]
One of the problems here is that we aren't able to discern what is usable and what is not for you.
Importing a huge collection of Raw files is a valid usecase of this feature.

We don't know that you're importing a folder which is an avatar etc.

It could probably be sorted with an option for "Batch Import ignore unusable" though.

**Comment by AmasterAmaster** [comment]
Hmm, one idea I had was making a popup dialog with a column of checkboxes, each representing a file type. The most common file types seen in Resonite would be at the top (or expected based on the number of files in a folder with that same file type). But that is more of a design problem I guess.

**Comment by lxw404** [comment]
This probably runs into the issue of being too monolithic for just a single use case.  https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/662 would probably better address this as you can just define your own way of handling this exact situation.

**Comment by AmasterAmaster** [comment]
@lxw404 It can extend from it yes, which would help new users out when it gets to that point. But focusing mainly on the folder imports themselves would also help the on-boarding process too. But very good to know.

**Comment by Dusty-Sprinkles** [comment]
Wouldn't an easy enough solution for something like this to just have advanced settings for the importer and make one of them "Import unsupported file types" and if you don't check that it skips over things resonite can't import?

That would at least solve the wall of .META files I think

Also this shouldn't be something a user has to design around like a modular system for #662 would be, because imo importing an avatar is still a lot lower level (arguably part of the "new user experience") than full on writing your own importer

**Comment by lxw404** [comment]
Agreed it shouldn't be something every user should design themselves, but https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/662 would allow such things to be distributed to any user from others who design importers.  Just adding yet another option to the importer might be wasted effort when it will be reworked anyways.

Overall though, trying to import an entire Unity directory shouldn't really be the way to go about importing Unity files in the first place tbh, it should probably be facilitated by: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/6.

**Comment by AmasterAmaster** [comment]
And even then, unity is going out the door, which would make that Unity SDK issue pointless... but I digress.

**Comment by lxw404** [comment]
https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/6 refers to importing content from Unity, (i.e. what you're trying to do by dragging Unity folders into Resonite).  In that issue, Froox directly addresses that due to the fact that we are removing Unity entirely, it will be designed as a way of translating Unity scenes without having to run Unity through FrooxEngine itself.

**Comment by ProbablePrime** [comment]
Thanks for the ideas, to be clear, I didn't say no. Mine was more "Hmmmm... how best to do".
